Library/Homebrew/metaformula.rb

```diff
@@ -0,0 +1,56 @@
+require "formula"
+
+# A meta-formula is a special kind of formula which will
+# * Allow bulk management of several related formulae (only `install `now)
+# * extend the semantics of 'dependency' to casks(TODO)
+#
+# As a result, {MetaFormula} differs from {Formula} in that
+# * `url` is not mandatory
+# * `version` is mandatory, (since it can not be deduced from `url` now)
+# * the keg directory can be empty now (which will influence the "empty installation" check in audit.rb)
+
+class MetaFormula < Formula
+  # Allow specs with no url
+  def spec_eval(name)
+    super
+
+    return unless name.equal? :stable
+    # we allow :stable spec with no url here
+    # unnecessary to keep three specs
+    # if we do, {Formula#installed_prefix} will report an error when `brew install xxx`
+    @stable = self.class.send(:stable)
+    @stable.owner = self
+    # `Formula#determine_active_spec` should now raise no "formulae require at least a URL" error
+  end
+
+  # Ignore url error in attribute validation
+  def validate_attributes!
+    super
+  rescue FormulaValidationError => e
+    raise unless e.attr.equal? :url
+  end
+
+  # Metaformula is not designed to be bottled for now
+  def pour_bottle?
```

Comment:
I'd consider setting `bottle_disabled` instead which will prevent it being bottled or installed from a bottle.